### PR TITLE
DATAES-363 fix existsById check

### PR DESCRIPTION
**Pull request by michael-wirth**
- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATAES).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

**Comment by michael-wirth** [comment]
I anyone reviewing these pull request? There isn't much going on in the last 2 months. Is this project on ice?

**Comment by michael-wirth** [comment]
any update on this?

**Comment by xhaggi** [comment]
Could you please rebase your branch against the master.

File: src/test/java/org/springframework/data/elasticsearch/repository/support/UUIDElasticsearchRepositoryTests.java [hunk 1]
```diff
@@ -262,6 +263,18 @@ public void shouldReturnTrueGivenDocumentWithIdExists() {
 		assertEquals(exist, true);
 	}
 
+	@Test // DATAES-363
+	public void shouldReturnFalseGivenDocumentWithIdDoesNotExist() {
+		// given
+		UUID documentId = UUID.randomUUID();
+
+		// when
+		boolean exist = repository.existsById(documentId);
```
Comment (xhaggi) [inline src/test/java/org/springframework/data/elasticsearch/repository/support/UUIDElasticsearchRepositoryTests.java]: should be `documentId.toString()` or better 

```java
// given
String documentId = UUID.randomUUID().toString();
```

Comment (xhaggi) [reply src/test/java/org/springframework/data/elasticsearch/repository/support/UUIDElasticsearchRepositoryTests.java]: sorry .. ignore my comment.

File: src/test/java/org/springframework/data/elasticsearch/repository/support/SimpleElasticsearchRepositoryTests.java [hunk 2]
```diff
@@ -273,6 +274,17 @@ public void shouldReturnTrueGivenDocumentWithIdExists() {
 		// then
 		assertEquals(exist, true);
 	}
+	@Test // DATAES-363
```
Comment (xhaggi) [inline src/test/java/org/springframework/data/elasticsearch/repository/support/SimpleElasticsearchRepositoryTests.java]: could you please add a new line before.

**Comment by michael-wirth** [comment]
done

**Comment by xhaggi** [comment]
@michael-wirth Thank you.
@akonczak could you please merge this PR and also backport it to 3.0.x

**Pull request closed without merging**